src/core/lib/iomgr/iocp_windows.cc

```diff
@@ -155,4 +163,20 @@ void grpc_iocp_add_socket(grpc_winsocket* socket) {
   GPR_ASSERT(ret == g_iocp);
 }
 
+void grpc_iocp_register_socket_shutdown(grpc_winsocket* socket) {
```

Comment:
Does this method need to be guarded by the state_mu lock? The lock is currently when calling this method, but `shutdown_registered` is not consistently protected across all uses.